MetaMask/metamask-extension
# feat(restore): Add Restore type to onboarding flow

## **Description**
This Pull Request adds a new onboarding 'firstTimeFlowType' called 'restore'. In a future PR (#23056) we will provide a path forward for users who suffer from a state corruption error, wherein their vault is backed up to local storage and they will be given an option to restore from that vault when the error occurs.

This PR adds a path through the onboarding screens for those users, skipping the step of choosing "Import" or "Create" and instead going straight to MetaMetrics opt-in.

This flow is **currently inaccessible** to users but I did add an E2E test to cover the flow by forcing state into the shape it would be in after a restore. 

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23739?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

Force MetaMask into the state required to test this:
```js
chrome.storage.local.get((state) => chrome.storage.local.set({ data: {
  ...state.data,
  OnboardingController: { completedOnboarding: false, firstTimeFlowType: 'restore', seedPhraseBackedUp: null },
  MetaMetricsController: { participateInMetaMetrics: null, metaMetricsId: null },
}}));
```

1. First, ensure you're on a local dev build with a working wallet initialized.
2. Open the background console from the extension management screen.
3. Use the steps above to get your state tree into a restore state.
4. Reload the extension from the extension management screen.
5. Open the extension and you should be requested to unlock your wallet with password. 
6. Your password should work.
7. You should see MetaMetrics optin screen and be able to continue onboarding

## **Screenshots/Recordings**


No changes, the only thing that changes is the skipping of the create/import step.

### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
